src/simulation/m_ibm.fpp

```diff
@@ -95,6 +95,35 @@ contains
         @:ALLOCATE(ghost_points(1:num_gps))
         @:ALLOCATE(inner_points(1:num_inner_gps))
 
+        ! Initialize the ip component of each ghost point
+        do i = 1, num_gps
+            allocate (ghost_points(i)%ip%alpha_rho(num_fluids))
+            allocate (ghost_points(i)%ip%alpha(num_fluids))
+            ghost_points(i)%ip%vel = 0.0_wp
+            ghost_points(i)%ip%pressure = 0.0_wp
+
+            if (surface_tension) then
+                ghost_points(i)%ip%c = 0.0_wp
+            end if
+
+            if (bubbles_euler) then
+                allocate (ghost_points(i)%ip%r(nb))
+                allocate (ghost_points(i)%ip%v(nb))
+                if (.not. polytropic) then
+                    allocate (ghost_points(i)%ip%pb(nb))
+                    allocate (ghost_points(i)%ip%mv(nb))
+                end if
+            end if
+
+            if (qbmm) then
+                allocate (ghost_points(i)%ip%nmom(nb*nmom))
```

Comment:
we use the `@:ALLOCATE` macro in almost all cases i think, since it declares the variable on the device.